sv.c

```diff
@@ -17729,6 +17717,43 @@ Perl_report_uninit(pTHX_ const SV *uninit_sv)
     GCC_DIAG_RESTORE_STMT;
 }
 
+/* This helper function for Perl_sv_setsv_flags is as cold as they come.
+ * We should almost never call it and it will definitely croak when we do.
+ * Therefore it should not matter that it is not close to the main function
+ * or that we make it redo work that the caller already did.
+ * The main aim is to keep Perl_sv_setsv_flags as slim as possible and this
+ * includes keeping the call sites for this function small.
+ */
+void S_sv_setsv_flags_croak(pTHX_ SV * const dsv, SV * const ssv)
+{
+    if (SvIS_FREED(dsv)) {
+        Perl_croak(aTHX_ "panic: attempt to copy value %" SVf
```

Comment:
Switch as many Perl_croak() to Perl_croak_nocontext() as possible if its in the same C fnc. This is panic code and almost unreachable. Drops all the copy/mov/push cpu instructions passing my_perl.

Reply:
Why would we want that? We don't do that with any other the many other PerlAPI calls either.

Reply:
Removing the my_perl arg on threaded perl from each croak() call site, drops 4-7 bytes of machine code, probable 8 or 12 bytes on ARM64. do it on 20 lines/call points. Now 100 bytes is gone from libperl, through a regexp substitute and very little thinking/time required by the author for a cheap fast (dev human time) little perf boost.

Assert branches/mem corruption panics printing branches are never perf critical, and almost unreachable for production perl code. And "perl code" calling PP ```die()``` does pass my_perl ptr through the ```Perl_pp_die()->Perl_die_sv()->Perl_croak_sv()->Perl_die_unwind()->longjmp()``` call chain, so we don't strip the my_perl ptr through PP die, Ive check in the past for bad Perl_get_contexts() there, PP_die() has no low level Perl specific perf issues IMO.  But for individual C API core die()/panics() format string, they are pretty heavy in Perl C. Every C level argument minimized to the the format string is better.

Excellent debugging info has a place, but perl doesnt use int error codes or error objects passed internally for its panic/semi asset messages. Most C/C++ code doesn't keep the ASCII text string support/debug code anywhere near average runtime code. The end user must explicitly pass that int error code or error object to a "error exception decoder" framework in another cold .o file in that app if they want an ASCII string description, 

Perl core traditionally keeps its fatal "text error formatter" code, right inside a C for loop, and perl often drops 4-8 pointers worth of "push &some_const_string_U32" format strings in machine code, and that overhead builds up if 2 64 bytes/8 ptr cache lines are 80% cold format C string args/getters funcs/misc, and 20% production statements on the edges of that 128 byte unit. Threaded perl is another layer of runtime overhead, but if the Perl C API, is designed a little bit smarter (PerlEnv vtable API needs a diet). Threaded perl/IMPLICT_CONTEXT/my_perl pointer builds, are just the "this" pointer in C++ machine code wise anywhere else. I've never seen anywhere in soft dev community complaints where someone wants to "optimize away" the ```this``` var in C/C++.

My point is CPU has to skip over format string machine code, and those branches never execute unless the process is about to SEGV, and nobody trap/resume execution after a ```bizzare copy of``` error. Aslong as ```Perl_pp_die()``` PP support written in C code, passes around my_perl, nobody can harm that their PP perl benchmark that measures units of "1000 exceptions per second" or "1000 die();s per second" was made worse. Perl_croak_nocontext() was originally a back compat fix for var args/CPreProc doesnt do var args, but IMO Perl_croak_nocontext() is little known tool that should be used more for reducing the "perl is slow" debate.

The user at runtime can spend an extra 2-4 microseconds and go back into the OS's TLS API and get back the my_perl ptr (Perl_get_context()) to finishing printing to stderr/expanding format strings, and the process is about to exit(1) or SEGV in a couple milliseconds anyways. 4 microseconds or 10 us of CPU or wall time can be spent, on a proc about to exit, in exchange for a permanent runtime benefit in primary execution flow/hot or warm code,  to all other global users.

Reply:
> Switch as many Perl_croak() to Perl_croak_nocontext() as possible

I think that if we want to do this, it should be a separate PR.